internal/index/indexscanner/scanner_test.go

```diff
@@ -91,53 +94,50 @@ func TestReadPluginFile(t *testing.T) {
 }
 
 func TestReadReceiptFile(t *testing.T) {
-	type args struct {
-		receiptFilePath string
-	}
 	tests := []struct {
-		name       string
-		args       args
-		wantErr    bool
-		matchFirst labels.Set
+		name            string
+		receiptFileName string
+		wantErr         bool
+		matchFirst      labels.Set
 	}{
 		{
-			name: "read receipt file",
-			args: args{
-				// TODO(chriskim06) use different file when new receipt fields are added
-				receiptFilePath: filepath.Join(testdataPath(t), "testindex", "plugins", "foo.yaml"),
-			},
-			wantErr: false,
+			name:            "read receipt file",
+			receiptFileName: "foo.yaml",
+			wantErr:         false,
 			matchFirst: labels.Set{
-				"os": "macos",
+				"arch": "amd64",
+				"os":   "linux",
 			},
 		},
 	}
 	neverMatch := labels.Set{}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got, err := ReadReceiptFromFile(tt.args.receiptFilePath)
+			tmpDir, cleanup := testutil.NewTempDir(t)
+			defer cleanup()
+
+			receipt := testutil.NewPlugin().WithName("foo").V()
+			receiptFile := filepath.Join(tmpDir.Root(), "receipts", tt.receiptFileName)
+			tmpDir.WriteYaml(receiptFile, receipt)
+
+			got, err := ReadReceiptFromFile(receiptFile)
 			if (err != nil) != tt.wantErr {
 				t.Errorf("ReadReceiptFromFile() error = %v, wantErr %v", err, tt.wantErr)
-				return
 			}
 			if err != nil {
 				t.Error(err)
-				return
 			}
-			if got.Name != "foo" && got.Kind != "Plugin" {
-				t.Errorf("ReadReceiptFromFile() has not parsed the metainformations %v", got)
-				return
+			if cmp.Diff(got.Plugin, receipt) != "" {
```

Comment:
`cmp.Diff` already formats the differences in an easy to read way. You should do
```go
if diff := cmp.Diff(got.Plugin, receipt); diff != "" {
   t.Errorf("Parsed receipt has unexpected content: %s", diff)
}
```